homeassistant/components/plugwise/util.py

```diff
@@ -3,13 +3,26 @@
 from collections.abc import Awaitable, Callable, Coroutine
 from typing import Any, Concatenate
 
+from packaging import version
 from plugwise.exceptions import PlugwiseException
 
 from homeassistant.exceptions import HomeAssistantError
 
+from .const import DEFAULT_TIMEOUT
 from .entity import PlugwiseEntity
 
 
+def get_timeout_for_version(version_str: str) -> int:
+    """Determine timeout value based on gateway version.
+
+    A gateway firmware version > 3.2.0 should mean a latest-generation-device, allowing for a timeout of 10s.
+    Legacy devices require a timeout of 30s.
+    """
+    if version.parse(version_str) >= version.parse("3.2.0"):
+        return 10
+    return DEFAULT_TIMEOUT
```

Comment:
I really think this is not integration logic.

The fact we pass down a timeout (which is also device depended) from Home Assistant down to the library is odd. Why not handle all of that in the library?

Home Assistant connects with a device... without passing a timeout, and it gets a timeout raised when... well whenever 🤷

Reply:
Yes, that's how we did it before but then we're stuck with an initial timeout of 30 secs for all devices.
So if a user make a typo she/he has to wait for 90 sec before she/he can correct and try again.

The nice thing of the HA discovery is that we can use the discovered information to set the timeout as early as possible, from within HA, to 10 sec for most of the user having non-legacy devices.
That was my thinking behind this PR.